knowledge-base/architectural-decisions-records/v4-api.md

```diff
@@ -217,27 +217,137 @@ The possible filter functions will depend on the data type. For example a string
+    },
+    {
+      "FirstName": "Suzanne",
+      "LastName": "Brown"
+    }
+  ]
+}
+```
+- delta update by using a `@delta` annotation
+  - delete entities (`@removed`)
+  - delete links (`reason = deleted`)
+  - add links (with id/key)
+  - update entities (with id/key + Partial<Entity>)
+  - create new entities + add links (without id/key)
+``` json
+{
+  "@type": "#Northwind.Manager",
+  "FirstName": "Patricia",
+  "DirectReports@delta": [
+    {
+      "@removed": {
+        "reason": "deleted"
+      },
+      "@id": "Employees(3)"
+    },
+    {
+      "@removed": {
+        "reason": "changed"
+      },
+      "@id": "Employees(4)"
+    },
+    {
+      "@id": "Employees(5)"
+    },
+    {
+      "@id": "Employees(6)",
+      "LastName": "Smith"
+    },
+    {
+      "FirstName": "Suzanne",
+      "LastName": "Brown"
+    }
+  ]
+}
 ```
-### Questions
-- Understand what deep update actually is
-- Refer to Alex to understand details
-- Consider differences between 4.0 and 4.01?
-
 ### Proposal(s)
-- Same as v2 create as child of
-  ```ts
-  TestEntityMultiLink.requestBuilder()
-    .update(multiLinkEntity)
-    .asChildOf(testEntity, TestEntity.TO_MULTI_LINK)
-  ```
+#### Full set
+- like deep create
+- no api changes, only behaviour changes
+- be aware of the `@id` field that needs to be handled
+   - [chosen] add `@id` attribute implicitly in the request builder (Java)
+   - add `@id` as default attribute to the abstract entity,
+   - [interesting][optional] change navi-property type to e.g., :
+   ```
+   toOtherMultiLink: (TestEntityOtherMultiLink & Identifiable)[];
+   
+   interface Identifiable{
+       _id: string
+   }
+   ```
+   - hint: `@id` is used [here](http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_LinktoRelatedEntitiesWhenCreatinganE)
+   
+``` typescript
+    const singleLink = getSingleLinkFromSystem();
+    singleLink.stringProperty = 'new value';
+    
+    const entity = TestEntity.builder()
+      .toSingleLink(singleLink)
+      .build();
+      
+    new UpdateRequestBuilder(TestEntity, entity)
+    .execute(destination);
+```
+#### Delta update
+``` typescript
+    new UpdateRequestBuilder(TestEntity, entity)
+      .deepUpdateWithDeltaPayload(
+      TestEntity.TO_MULTI_LINK, deltaMultiLink1, deltaMultiLink2
+    ).deepUpdateWithDeltaPayload(
+      someOtherNaviProperties...
+    )
+    .execute(destination);
+```
+```
+deltaMultiLink: TestEntityMultiLink & IsRemoved;
+
+interface IsRemoved{
+    removed?: boolean;
+    reason?: string;
+}
+```
+
+Again, `@id` can be treated the same way as the full set
 
 ### Decision
-- TBD, decision postponed after we understand details
+#### 4.0 V.S. 4.0.1
+We stick to version `4.0.1`, since it's newer, the latest and more powerful.
+Also, this [link](http://docs.oasis-open.org/odata/new-in-odata/v4.01/cn01/new-in-odata-v4.01-cn01.html#_Toc485385071)
+indicates that 4.0.1 introduces deep updates.
+### Scope
+- Full set (Y)
```

Comment:
```suggestion
We will do the the implementation of the `full set` and not the delta since there is no test system to support it.
- Full set (Y)
```